src/Sonarr.Api.V3/Indexers/ReleasePushModule.cs

```diff
@@ -1,50 +1,111 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
 using FluentValidation;
+using FluentValidation.Results;
 using Nancy;
 using Nancy.ModelBinding;
 using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore;
 using NzbDrone.Core.DecisionEngine;
 using NzbDrone.Core.Download;
+using NzbDrone.Core.Indexers;
 using NzbDrone.Core.Parser.Model;
 using Sonarr.Http.Extensions;
+using Sonarr.Http.REST;
 
 namespace Sonarr.Api.V3.Indexers
 {
     class ReleasePushModule : ReleaseModuleBase
     {
         private readonly IMakeDownloadDecision _downloadDecisionMaker;
         private readonly IProcessDownloadDecisions _downloadDecisionProcessor;
+        private readonly IIndexerFactory _indexerFactory;
         private readonly Logger _logger;
+        private ResourceValidator<ReleaseResource> _releaseValidator;
 
         public ReleasePushModule(IMakeDownloadDecision downloadDecisionMaker,
                                  IProcessDownloadDecisions downloadDecisionProcessor,
+                                 IIndexerFactory indexerFactory,
                                  Logger logger)
         {
             _downloadDecisionMaker = downloadDecisionMaker;
             _downloadDecisionProcessor = downloadDecisionProcessor;
+            _indexerFactory = indexerFactory;
             _logger = logger;
+;
+            _releaseValidator = new ResourceValidator<ReleaseResource>();
+            _releaseValidator.RuleFor(s => s.Title).NotEmpty();
```

Comment:
What's the idea behind replacing PostValidator with this?

Reply:
PostValidator doesn't work unless you `CreateResource`, which would require a ID response that then gets the resource, which would be more complicated than we wanted here.

Reply:
Right, so the original code using the Post[..] = never even used the validator.

Shouldn't we use 'ReadResourceFromRequest' instead of this.Bind in that case? That would invoke the Validator while still allowing a custom result. Similar to what we do for Post["test"] and Post["action/{action}"].

Reply:
Yeah, that's a better option, didn't think about. Now that that's out of the way 😄